### PR TITLE
command options : Add rpc flag

### PR DESCRIPTION
**Pull request by hyochan-brown**
## Proposed changes
Enhanced rpc flags compatibility with Ethereum
```
./ken --rpc --rpcapi klay
WARN[06/14,16:18:00 +09] [18] The flag --rpc is deprecated and will be removed in the future, please use --http 
WARN[06/14,16:18:00 +09] [18] The flag --rpcapi is deprecated and will be removed in the future, please use --http.api 
INFO[06/14,16:18:00 +09] [18] Cypress network ID is set                 networkid=8217
```
```
./ken --http --http.api klay
INFO[06/14,16:18:00 +09] [18] Cypress network ID is set                 networkid=8217
```

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by aidan-kwon** [comment]
@hyochan-brown Could you change deamon files such as `build/packaging/linux/bin/kcnd` not to use deprecated options.

File: build/packaging/linux/conf/kcnd.conf [hunk 1]
```diff
@@ -22,12 +22,12 @@ TXPOOL_NONEXEC_SLOTS_ACCOUNT=16384
 TXPOOL_LIFE_TIME="5m"
 
 # rpc options setting
-RPC_ENABLE=0 # if this is set, the following options will be used
-RPC_API="klay" # available apis: admin,debug,klay,miner,net,personal,rpc,txpool,web3
-RPC_PORT=8551
-RPC_ADDR="0.0.0.0"
-RPC_CORSDOMAIN="*"
-RPC_VHOSTS="*"
+HTTP_ENABLE=0 # if this is set, the following options will be used
```
Comment (jeongkyun-oh) [inline build/packaging/linux/conf/kcnd.conf:25]: If this config is updated, then it may cause problems with the old configuration. Don't we need backward compatibility for this?

**Review by jeongkyun-oh — APPROVED** [review]

File: cmd/utils/nodecmd/flags_test.go [hunk 2]
```diff
@@ -375,39 +375,39 @@ var flagsWithValues = []struct {
 		errors:      []int{ErrorInvalidValue, NonError, ErrorInvalidValue},
 	},
 	{
-		flag:     "--rpc",
+		flag:     "--http",
 		flagType: FlagTypeBoolean,
 	},
 	{
-		flag:        "--rpcaddr",
+		flag:        "--http.addr",
```
Comment (jeongkyun-oh) [inline cmd/utils/nodecmd/flags_test.go:382]: It is minor, but it is better to leave rpc tests and add http tests instead of substitution.

File: cmd/kbn/config.go [hunk 3]
```diff
@@ -219,24 +219,51 @@ func setAuthorizedNodes(ctx *cli.Context, cfg *bootnodeConfig) {
 // setHTTP creates the HTTP RPC listener interface string from the set
 // command line flags, returning empty if the HTTP endpoint is disabled.
 func setHTTP(ctx *cli.Context, cfg *bootnodeConfig) {
-	if ctx.GlobalBool(utils.RPCEnabledFlag.Name) && cfg.HTTPHost == "" {
+	if ctx.GlobalBool(utils.HTTPEnabledFlag.Name) && cfg.HTTPHost == "" {
```
Comment (jeongkyun-oh) [inline cmd/kbn/config.go:222]: I put comments on another PR, but do you have any better way to check legacy flags?

Comment (hyochan-brown) [reply cmd/kbn/config.go]: I reflected it in this PR too! 
Thanks JK!

File: build/packaging/linux/bin/kcnd [hunk 4]
```diff
@@ -74,38 +74,59 @@ start() {
...
+        fi
+        if [ ! -z $HTTP_CORSDOMAIN ]; then
+            OPTIONS="$OPTIONS --http.corsdomain $HTTP_CORSDOMAIN"
+        fi
+        if [ ! -z $HTTP_VHOSTS ]; then
+            OPTIONS="$OPTIONS --http.vhosts $HTTP_VHOSTS"
+        fi
+    fi
+
+    #Deprecated options
```
Comment (yoomee1313) [inline build/packaging/linux/bin/kcnd:96]: @hyochan-brown Can I ask why you remain deprecated options?

File: cmd/utils/flags.go [hunk 5]
```diff
@@ -108,7 +108,7 @@ var (
 	// TODO-Klaytn-Bootnode: redefine networkid
 	NetworkIdFlag = cli.Uint64Flag{
 		Name:  "networkid",
-		Usage: "Network identifier (integer, 1=MainNet (Not yet launched), 1000=Aspen, 1001=Baobab)",
+		Usage: "Network identifier (integer, 8217=MainNet, , 1001=Baobab)",
```
Comment (yoomee1313) [inline cmd/utils/flags.go:111]: `Mainnet` is not changed to `Cypress`. Can I ask why it is not changed to `Cypress`?

File: cmd/kbn/config.go [hunk 6]
```diff
@@ -280,11 +284,15 @@ func setgRPC(ctx *cli.Context, cfg *bootnodeConfig) {
 		cfg.GRPCHost = "127.0.0.1"
 		if ctx.GlobalIsSet(utils.GRPCListenAddrFlag.Name) {
 			cfg.GRPCHost = ctx.GlobalString(utils.GRPCListenAddrFlag.Name)
+		} else if ctx.GlobalIsSet(utils.LegacyGRPCListenAddrFlag.Name) {
+			cfg.GRPCHost = ctx.GlobalString(utils.LegacyGRPCListenAddrFlag.Name)
```
Comment (yoomee1313) [inline cmd/kbn/config.go:288]: It seems that the `getFlagName` job is also needed at here. Could you check it once more? And it seems same codes are exists in multiple files

File: cmd/utils/flags.go [hunk 7]
```diff
@@ -541,6 +541,97 @@ var (
...
+		Value: "",
+	}
+	LegacyWSMaxSubscriptionPerConn = cli.IntFlag{
+		Name:  "wsmaxsubscriptionperconn",
+		Usage: "Allowed maximum subscription number per a websocket connection",
+		Value: int(rpc.MaxSubscriptionPerWSConn),
+	}
+	LegacyWSReadDeadLine = cli.Int64Flag{
+		Name:  "wsreaddeadline",
+		Usage: "Set the read deadline on the underlying network connection in seconds. 0 means read will not timeout (deprecated, use --ws.readdeadline)",
```
Comment (yoomee1313) [inline cmd/utils/flags.go:611]: Can the rpc flags not specify the version to be deprecated?

File: build/packaging/linux/conf/kcnd.conf (hunk 1 above)
Comment (winnie-byun) [reply build/packaging/linux/conf/kcnd.conf:25]: I don't think we need to backward compatibility for config files. Rather we could leave a comment in the file : `This configuration options are for v1.6.4 and upper`.
If you are going to make backward compatibility, this line could be changed to : 
```
RPC_ENABLE=0  # for v1.6.3 and under. Same with HTTP_ENABLE
HTTP_ENABLE=0 # for v1.6.4 and upper. If this is set, the following options will be used
```

File: cmd/utils/flags.go (hunk 5 above)
Comment (hyochan-brown) [reply cmd/utils/flags.go]: I reflected it. Thanks!

File: cmd/kbn/config.go (hunk 6 above)
Comment (hyochan-brown) [reply cmd/kbn/config.go]: I reflected it. Thanks!

File: cmd/utils/flags.go (hunk 7 above)
Comment (hyochan-brown) [reply cmd/utils/flags.go]: I'll reflect it. Thanks!

File: build/packaging/linux/bin/kcnd (hunk 4 above)
Comment (hyochan-brown) [reply build/packaging/linux/bin/kcnd:96]: I heard that users don't change the config file when update the daemon.
I remain it in the old config file compatibility.

Comment (yoomee1313) [reply build/packaging/linux/bin/kcnd:96]: Thanks, it helps a lot to understand!

File: cmd/utils/flags.go [hunk 8]
```diff
@@ -1749,3 +1852,14 @@ func getNetworkId(ctx *cli.Context) (uint64, bool) {
 		return params.CypressNetworkId, false
 	}
 }
+
+func getFlagName(ctx *cli.Context, flag cli.Flag, legacyFlag cli.Flag, deprecateVersion string) string {
```
Comment (yoomee1313) [inline cmd/utils/flags.go:1856]: @hyochan-brown It seems that idential `getFlagName` is defined in cmd/kbn/config.go.. Could you check this?

File: cmd/utils/flags.go [hunk 9]
```diff
@@ -1192,11 +1291,15 @@ func setgRPC(ctx *cli.Context, cfg *node.Config) {
 		cfg.GRPCHost = "127.0.0.1"
 		if ctx.GlobalIsSet(GRPCListenAddrFlag.Name) {
 			cfg.GRPCHost = ctx.GlobalString(GRPCListenAddrFlag.Name)
+		} else if ctx.GlobalIsSet(LegacyGRPCListenAddrFlag.Name) {
```
Comment (yoomee1313) [inline cmd/utils/flags.go:1294]: It is minor comment, but could you apply `getFlagName` at GRPCListenAddr and GRPCPort too?

File: cmd/utils/flags.go (hunk 8 above)
Comment (hyochan-brown) [reply cmd/utils/flags.go:1856]: It was just made to remove the duplicate code, I thought it didn't have to be open in the external package.
So I added it on both flags.go / config.go... 
Is it better to open it on the external package to remove deduplication?

**Comment by yoomee1313** [comment]
This PR is closed. Previous work will be moved to a new branch and resumed later with a new PR. https://github.com/yoomee1313/klaytn/pull/new/feature/add-rpc-flags
